admission-webhook/deploy/deploy-gmsa-webhook.sh

```diff
@@ -150,6 +146,14 @@ main() {
         effect: NoSchedule'
     fi
 
+    if [ -x "$(command -v envsubst)" ]; then
+        ENVSUBST="$(command -v envsubst) < $TEMPLATE_PATH"
+    elif [ -x "$(command -v docker)" ]; then
+        ENVSUBST="docker run --rm -v $TEMPLATE_PATH:$TEMPLATE_PATH -eTLS_PRIVATE_KEY -eTLS_CERTIFICATE -eCA_BUNDLE -eRBAC_ROLE_NAME -eNAME -eNAMESPACE -eIMAGE_NAME -eTOLERATIONS supinf/envsubst $TEMPLATE_PATH"
```

Comment:
I'm sorry, I didn't know that `env` does not play nice with multi-lines env variables, my apologies :(

That being said, it's still possible to not have this unsightly list of variables list to maintain here either, something like
```bash
maggy:tmp wk$ cat target.yml
foo: ${FOO}
bar: ${BAR}
maggy:tmp wk$ FOO="multi                                                     
line
foo
value
"
maggy:tmp wk$ CMD='envsubst < target.yml'
maggy:tmp wk$ env - PATH="$PATH" FOO="$FOO" BAR='bar value' "$BASH" -c "$CMD"
foo: multi
line
foo
value

bar: bar value
maggy:tmp wk$ CMD='EXIT_STATUS=0; TMP_FILE=$(mktemp) && export -p > "$TMP_FILE" && docker run --rm -v $(pwd):/tmp -e TMP_FILE -v "$TMP_FILE:$TMP_FILE" wk88/envsubst /tmp/target.yml "$TMP_FILE" || EXIT_STATUS=$? ; rm -f "$TMP_FILE" ; exit $EXIT_STATUS'
maggy:tmp wk$ env - PATH="$PATH" FOO="$FOO" BAR='bar value' "$BASH" -c "$CMD"
foo: multi
line
foo
value

bar: bar value
```
seems to do the trick.

Up to you to see what you prefer :) ?

Reply:
Also, at this point, we might want to add a Travis job to check that deploying without envsubst works, since that breaking would break k8s' CI in a way that might be a bit painful to track down.

Would be as simple as adding a new job to https://github.com/kubernetes-sigs/windows-gmsa/blob/master/admission-webhook/.travis.sh#L12-L23 that would remove envsubst prior to running, then run integration tests.

Reply:
Rather than having all of those commands stored in a variable, I'd write a separate function for generating the manifests file, which would make it a whole lot easier to read/understand, if that's ok with you.

Reply:
Of course :) Thanks!

Reply:
Would it be ok if I do the travis test changes in a separate commit?

Reply:
Sure, though it should really be just a few lines, up to you.